===================================================================
RCS file: /cvs/glibc/libc/elf/dl-load.c,v
retrieving revision 1.155
retrieving revision 1.156
diff -u -r1.155 -r1.156
--- libc/elf/dl-load.c	2000/12/27 20:18:52	1.155
+++ libc/elf/dl-load.c	2001/01/09 04:26:00	1.156
@@ -1585,7 +1585,7 @@
 			    &loader->l_runpath_dirs, &realname, &fb);
 	}
 
-      if (fd == -1)
+      if (fd == -1 && (! preloaded || ! __libc_enable_secure))
 	{
 	  /* Check the list of libraries in the file /etc/ld.so.cache,
 	     for compatibility with Linux's ldconfig program.  */
===================================================================
RCS file: /cvs/glibc/libc/elf/dl-profile.c,v
retrieving revision 1.21
retrieving revision 1.22
diff -u -r1.21 -r1.22
--- libc/elf/dl-profile.c	2000/12/27 19:55:23	1.21
+++ libc/elf/dl-profile.c	2001/01/09 03:08:28	1.22
@@ -256,7 +256,12 @@
   *cp++ = '/';
   __stpcpy (__stpcpy (cp, _dl_profile), ".profile");
 
-  fd = __open (filename, O_RDWR | O_CREAT, 0666);
+#ifdef O_NOFOLLOW
+# define EXTRA_FLAGS | O_NOFOLLOW
+#else
+# define EXTRA_FLAGS
+#endif
+  fd = __open (filename, O_RDWR | O_CREAT EXTRA_FLAGS, 0666);
   if (fd == -1)
     {
       /* We cannot write the profiling data so don't do anything.  */
===================================================================
RCS file: /cvs/glibc/libc/elf/rtld.c,v
retrieving revision 1.187
retrieving revision 1.188
diff -u -r1.187 -r1.188
--- libc/elf/rtld.c	2000/12/31 18:03:27	1.187
+++ libc/elf/rtld.c	2001/01/09 04:32:16	1.188
@@ -1322,7 +1322,7 @@
   char *debug_output = NULL;
 
   /* This is the default place for profiling data file.  */
-  _dl_profile_output = "/var/tmp";
+  _dl_profile_output = __libc_enable_secure ? "/var/profile" : "/var/tmp";
 
   while ((envline = _dl_next_ld_env_entry (&runp)) != NULL)
     {
